Update profile

PaulMicu opened the pull request:
user should be able to update his old password to a new one, it should also be able to change the displayed profile picture and it should be able to add or change a user bio

miche-cho committed:
changed edit page

changed the MySQLrequest function for updateProfile

added column user_bio into user table in the database

fixed typo in reqregister.php

PinMeister committed:
User bio update

Somewhat functional ability to update the user bio on the database.

Commit:
change password

fixed change password functiona;lity by moving the code in the correct file and writing the code. it checks for old password being good and checks if the two new password match. does not check if new password are "good"

MrWorme commented:
change password has been incorporated in the last commit, it needs a small modification to check the regex to make sure the new password is conform to the specs

miche-cho committed:
Code clean up

Cleaned up the tabmenu code
moved SQL queries to MySQLrequests as functions

This will make tabmenu cleaner and not as cluttered

miche-cho committed:
Profile page displays number of posts made, and post table refactoring

The user profile page now displays the  number of posts made by the user

a query and a getter function are made on the MySQLrequests page and the profilePageAction respectively.

Links leading to user profile pages are removed from the Post Table page

miche-cho committed:
Add Clean Code PHP

Added Jupeter's Clean Code PHP GitHub repo to our own, this is the clean coding style we are adopting for our php code

PaulMicu committed:
dynamic header info

made the information displayed when pressed on the icon to be dynamic to the user

PaulMicu committed:
modify picture

added option to change profile pic from a set of images and made those display proprely

PaulMicu committed:
Update constants.php

PaulMicu committed:
Merge branch 'update-profile' of https://github.com/milaroisin/SOEN341 into update-profile

PaulMicu committed:
Merge branch 'master' into update-profile
